res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -295,96 +353,87 @@ PioneerDDJSX.init = function(id) {
-        [1 / 8, 1 / 4, 1 / 2, 1, 2, 4, 8, 16],
-        [1 / 16, 1 / 8, 1 / 4, 1 / 2, 1, 2, 4, 8],
-        [1 / 32, 1 / 16, 1 / 8, 1 / 4, 1 / 2, 1, 2, 4]
-    ];
-
-    PioneerDDJSX.setNonDeckSoftTakeover(true);
-    PioneerDDJSX.bindNonDeckControlConnections(true);
-    PioneerDDJSX.initDeck("[Channel1]");
-    PioneerDDJSX.initDeck("[Channel2]");
-    PioneerDDJSX.initDeck("[Channel3]");
-    PioneerDDJSX.initDeck("[Channel4]");
-
-    // show Sampler Load configuration file dialog:
-    if (PioneerDDJSX.samplerShowLoadDialog) {
-        script.toggleControl("[Sampler]", "LoadSamplerBank");
-    }
-
     // set 32 Samplers as default:
     engine.setValue("[Master]", "num_samplers", 32);
 
-    // switch on Effect Units as default:
-    for (var index in PioneerDDJSX.fxUnitGroups) {
-        if (PioneerDDJSX.fxUnitGroups.hasOwnProperty(index)) {
-            engine.setValue(index, "enabled", true);
-        }
-    }
-
     // activate vu meter timer for Auto DJ:
     if (PioneerDDJSX.twinkleVumeterAutodjOn) {
         PioneerDDJSX.vuMeterTimer = engine.beginTimer(200, "PioneerDDJSX.vuMeterTwinkle()");
     }
 
-    // initialize autoDJ tick timer:
-    PioneerDDJSX.autoDJTickTimer = 0;
+    // send init message to controller:
+    midi.sendSysexMsg(PioneerDDJSX.initSendMsg, PioneerDDJSX.initSendMsg.length);
+    PioneerDDJSX.initTimer = engine.beginTimer(1000, "PioneerDDJSX.initTimerTick()");
 };
 
 PioneerDDJSX.shutdown = function() {
-    PioneerDDJSX.setNonDeckSoftTakeover(false);
-    PioneerDDJSX.bindNonDeckControlConnections(false);
     PioneerDDJSX.resetDeck("[Channel1]");
     PioneerDDJSX.resetDeck("[Channel2]");
     PioneerDDJSX.resetDeck("[Channel3]");
     PioneerDDJSX.resetDeck("[Channel4]");
 
     PioneerDDJSX.resetNonDeckLeds();
-
-    // stop vu meter timer for AutoDJ:
-    /* if (PioneerDDJSX.twinkleVumeterAutodjOn && PioneerDDJSX.vuMeterTimer) {
-         engine.stopTimer(PioneerDDJSX.vuMeterTimer);
-    } */
-    //not needed
-
-    // stop AutoDJ tick timer:
-    /* if (PioneerDDJSX.autoDJTickTimer) {
-		engine.stopTimer(PioneerDDJSX.autoDJTickTimer);
-	} */
-    //not needed
 };
 
 PioneerDDJSX.setDefaultSpeedSliderRange = function(group, range) {
     engine.setValue(group, "rateRange", range);
 };
 
+PioneerDDJSX.inboundSysex = function(data, length) {
+    engine.stopTimer(PioneerDDJSX.initTimer);
+    PioneerDDJSX.initTimer = 0;
+
+    if (length !== PioneerDDJSX.initRecvMsg.length) {
+        print("Sysex message has different length!");
+        return;
+    }
+
+    for (var i = 0; i < length; i++) {
+        if (data[i] !== PioneerDDJSX.initRecvMsg[i]) {
+            print("Init - Sysex handshake failed!");
+            return;
+        }
+    }
+
+    print("Init - Sysex handshake success!");
+    PioneerDDJSX.initAfterHandshake(true);
+};
+
+PioneerDDJSX.initAfterHandshake = function(status) {
+    // initiate control status request:
+    if (status) {
+        midi.sendShortMsg(0x9B, 0x08, 0x7F);
```

Comment:
Is it necessary to send the sysex message? Could you just send `0x9B, 0x08, 0x7F`?